osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/firmware/UpdateFirmwareCommandExecutorIntegrationTest.java

```diff
@@ -50,94 +50,178 @@ class UpdateFirmwareCommandExecutorIntegrationTest {
+              + "5fa821c678e71c05c47e1c69c4bfffe1fd");
+  final String fwFileHash = "951948459d2b7b59883cfc75c2bc2b7e4a0232ae7973a0d99526afd9458b0c86";
+  final byte[] firmwareFileMbus =
+      org.bouncycastle.util.encoders.Hex.decode(
+          "534d523500230011004000310000001000020801e91effffffff500303000000000000831c9d5aa5b4f"
+              + "fbfd057035a8a7896a4abe7afa36687fbc48944bcee0343eed3a75aab882ec1cf57820adfd4394e262d"
+              + "5fa821c678e71c05c47e1c69c4bfffe1fd");
 
   @BeforeEach
   void setUp() {
+
+    final int verificationStatusCheckInterval = 1;
+    final int verificationStatusCheckTimeout = 2;
+    final int initiationStatusCheckInterval = 3;
+    final int initiationStatusCheckTimeout = 4;
+
+    this.messageMetadata = MessageMetadata.newBuilder().withCorrelationUid("123456").build();
+
+    final ImageTransferProperties imageTransferProperties =
+        new ImageTransfer.ImageTransferProperties();
+    imageTransferProperties.setVerificationStatusCheckInterval(verificationStatusCheckInterval);
+    imageTransferProperties.setVerificationStatusCheckTimeout(verificationStatusCheckTimeout);
+    imageTransferProperties.setInitiationStatusCheckInterval(initiationStatusCheckInterval);
+    imageTransferProperties.setInitiationStatusCheckTimeout(initiationStatusCheckTimeout);
+
+    this.commandExecutor =
+        new UpdateFirmwareCommandExecutor(
+            this.dlmsDeviceRepository,
+            this.firmwareFileCachingRepository,
+            this.firmwareImageIdentifierCachingRepository,
+            this.macGenerationService,
+            imageTransferProperties);
+  }
+
+  private void initConnectionStubForTransfer(
+      final int firstTransferState, final int firstNotTransferredBlockNumber) {
+
     this.connectionStub = new DlmsConnectionStub();
     this.connectionManagerStub = new DlmsConnectionManagerStub(this.connectionStub);
 
     this.connectionStub.setDefaultReturnValue(DataObject.newArrayData(Collections.emptyList()));
 
+    if (firstTransferState > -1) {
+      this.connectionStub.addReturnValue(
+          this.createAttributeAddressForImageTransfer(ImageTransferAttribute.IMAGE_TRANSFER_STATUS),
+          DataObject.newInteger32Data(firstTransferState));
+    }
+
     this.connectionStub.addReturnValue(
         this.createAttributeAddressForImageTransfer(ImageTransferAttribute.IMAGE_TRANSFER_ENABLED),
         DataObject.newBoolData(true));
     this.connectionStub.addReturnValue(
         this.createAttributeAddressForImageTransfer(ImageTransferAttribute.IMAGE_TRANSFER_STATUS),
         DataObject.newInteger32Data(1),
-        5);
-    this.connectionStub.addReturnValue(
-        this.createAttributeAddressForImageTransfer(ImageTransferAttribute.IMAGE_TRANSFER_STATUS),
-        DataObject.newInteger32Data(6));
+        4);
+
+    // configure meter simulator to follow either FromFirstBlock or ResumeOnBlocks flow
```

Comment:
Using `simulator` here might be confusing since the connectionstub is used instead of a simulator.

Reply:
_configure the connection stub to follow ..._